packages/@lwc/integration-karma/test/custom-elements/index.spec.js

```diff
@@ -37,54 +38,93 @@ const undefinedElementError = /(Illegal constructor|does not define a custom ele
-            { tag: 'x-nonce2', Component: Nonce2 },
-        ];
-
-        // There are two ways that customElements.define can eventually get called:
-        // 1) createElement
-        // 2) explicit customElements.define
-        const creators = [
-            {
-                method: 'using createElement',
-                create: (tag, Component) => {
-                    return createElement(tag, { is: Component });
-                },
-            },
-            {
-                method: 'using CustomElementConstructor',
-                create: (tag, Component) => {
-                    customElements.define(tag, Component.CustomElementConstructor);
-                    return document.createElement(tag);
-                },
-            },
-        ];
-
-        creators.forEach(({ method, create }, i) => {
-            const { tag, Component } = nonceElements[i];
-
-            // For LWC components, the PivotCtor is exposed externally but not the UserCtor
-            it(method, () => {
-                expect(customElements.get(tag)).toBeUndefined();
-                const promise = customElements.whenDefined(tag);
-                expect(customElements.get(tag)).toBeUndefined();
-                const elm = create(tag, Component);
-                const Ctor = customElements.get(tag);
-                expect(typeof Ctor).toEqual('function');
+        it('using CustomElementConstructor', () => {
+            // Nonce elements should be defined only once in the entire Karma test suite
+            const tagName = 'x-nonce1';
+            expect(customElements.get(tagName)).toBeUndefined();
+            const promise = customElements.whenDefined(tagName);
+            expect(customElements.get(tagName)).toBeUndefined();
+            customElements.define(tagName, Nonce1.CustomElementConstructor);
+            const elm = document.createElement(tagName);
+            const Ctor = customElements.get(tagName);
+            expect(Ctor).toBe(Nonce1.CustomElementConstructor);
+            document.body.appendChild(elm);
+            expect(elm.expectedTagName).toEqual(tagName);
+            return promise
+                .then((Ctor) => {
+                    expect(Ctor).toBe(Nonce1.CustomElementConstructor);
+                    return customElements.whenDefined(tagName);
+                })
+                .then((Ctor) => {
+                    expect(Ctor).toBe(Nonce1.CustomElementConstructor);
+                });
+        });
+
+        if (window.lwcRuntimeFlags.ENABLE_SCOPED_CUSTOM_ELEMENT_REGISTRY) {
+            it('using lwc.createElement', () => {
+                const tagName = 'x-nonce2';
+                expect(customElements.get(tagName)).toBeUndefined();
+                let resolvedCtor;
+                const promise = customElements.whenDefined(tagName).then((ctor) => {
+                    resolvedCtor = ctor;
+                });
+                expect(customElements.get(tagName)).toBeUndefined();
```

Comment:
Between L66 and L71, the customElements registry is untouched except for registering the callback. In which case is it expected that `customElements.get(tagName)` will behave differently at L71?

Reply:
No you're right, this line is unnecessary.